plugin/input/journalctl/offset.go

```diff
@@ -0,0 +1,64 @@
+package journalctl
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+type offsetInfo struct {
+	current int64
+	offset  int64
+	cursor  string
+	path    string
+}
+
+func newOffsetInfo(path string) *offsetInfo {
+	return &offsetInfo{
+		path: path,
+	}
+}
+
+func (o *offsetInfo) set(cursor string) {
+	o.cursor = cursor
+	o.offset++
+}
+
+func (o *offsetInfo) loadFile() error {
+	file, err := os.Open(o.path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		} else {
+			return err
+		}
+	}
+	defer file.Close()
+
+	return o.load(file)
+}
+
+func (o *offsetInfo) load(r io.Reader) error {
+	if _, err := fmt.Fscanf(r, "- %d: %q", &o.offset, &o.cursor); err != nil {
+		return err
+	}
+	o.current = o.offset + 1
+	return nil
+}
+
+func (o *offsetInfo) saveFile() error {
+	file, err := os.Create(o.path)
```

Comment:
The file should be opened once and use the descriptor to write new information. Reopening file on each log message is bad